app/updatemanager.go

```diff
@@ -386,6 +409,81 @@ func (c *ControlMapPool) ClearExpired() {
 		},
 	)
 	c.Pool = newPool
+
+	c.saveToStore()
+}
+
+type controlMapPoolDbFormat struct {
+	Active  []*UpdateControlMap `json:"active"`
+	Expired []*UpdateControlMap `json:"expired"`
+}
+
+// Must only be called from functions that have already locked the mutex.
+func (c *ControlMapPool) saveToStore() {
+	if c.store == nil {
+		// Saving not enabled. This should not happen outside tests.
+		return
+	}
+
+	active, expired := c.getAll()
+
+	toSave := controlMapPoolDbFormat{
+		// Intentionally not using label assignment here. We want to
+		// know if we're missing any fields.
+		active,
+		expired,
+	}
+
+	data, err := json.Marshal(&toSave)
```

Comment:
Idea:
Why not add a method to the `dbstore`, which looks something like:
```
Write(name string, m json.Marshaler)
```
and then add a `MarshalJSON` to the `controlMapDbFormat`, and writing can be:

```
db.Write(datastore.UpdateControlMaps, maps)
```

Reply:
I like the abstraction, but I think it will not save a lot, at least not as long as we're only marshalling pure JSON, and only once. It's just moving the code to a `MarshallJSON` function, but it's still the same code. Since I also have to add it to the other store types then, I think I will save this for another day.

Reply:
Yeh, it's a little out of scope :)

I have a feeling we're doing this a lot of place throughout our codebase though, like don't we always store JSON basically in the DB? :p